scripts/irods/test/test_istream.py

```diff
@@ -35,3 +36,47 @@ def test_istream_empty_file(self):
         finally:
             if os.path.exists(filename):
                 os.unlink(filename)
+
+    def test_creating_data_objects_using_replica_numbers_is_not_allowed__issue_5107(self):
+        data_object = 'cannot_create_data_objects'
+
+        # Show that the targeted data object does not exist.
+        self.admin.assert_icommand(['ils', '-l', data_object], 'STDERR', ['does not exist or user lacks access permission'])
+
+        # Show that attempts to write to a non-existent data object using replica numbers does
+        # not result in creation of the data object.
+        self.admin.assert_icommand_fail(['istream', 'write', '-n0', data_object], input='some data')
+        self.admin.assert_icommand_fail(['istream', 'write', '-n4', data_object], input='some data')
```

Comment:
assert the specific error seen?

Reply:
not sure how to do that with the `input=` parameter, but will investigate.

Reply:
ah, gotcha.